### PR TITLE
Backlog 16164 fix vanity

### PR DESCRIPTION
**Pull request by AKarmanov**
## JIRA

https://jira.jahia.org/browse/BACKLOG-16164

## Description

Fix vanity url generation

### CODE REVIEW

File: src/main/java/org/jahia/modules/sitemap/utils/VanityUrls.java [hunk 1]
```diff
@@ -55,6 +56,10 @@ public static String getActiveUrl(JCRNodeWrapper node, String langCode) throws R
         return getActiveUrl(vanityUrls);
     }
 
+    public static boolean canUseVanity() {
+        return Boolean.parseBoolean(SettingsBean.getInstance().getString("urlRewriteSeoRulesEnabled", "false"));
```
Comment (gflores-jahia) [inline src/main/java/org/jahia/modules/sitemap/utils/VanityUrls.java:60]: I think there's an equivalent `isUrlRewriteSeoRulesEnabled()` to avoid the boolean conversion but this works as well.

File: src/main/java/org/jahia/modules/sitemap/utils/VanityUrls.java [hunk 2]
```diff
@@ -55,6 +56,10 @@ public static String getActiveUrl(JCRNodeWrapper node, String langCode) throws R
         return getActiveUrl(vanityUrls);
     }
 
+    public static boolean canUseVanity() {
```
Comment (gflores-jahia) [inline src/main/java/org/jahia/modules/sitemap/utils/VanityUrls.java:59]: we should probably rename as this is unrelated to vanity settings. I suggest maybe `urlRewriteEnabled()` or along those lines instead. And since this is unrelated to Vanity URLs also would make sense  to move it probably to the more generic Utils class instead.

File: src/main/resources/META-INF/sitemap.tld [hunk 3]
```diff
@@ -26,6 +26,11 @@
         <function-class>org.jahia.modules.sitemap.utils.VanityUrls</function-class>
         <function-signature>boolean hasActiveUrl(org.jahia.services.content.JCRNodeWrapper)</function-signature>
     </function>
+    <function>
+        <name>canUseVanity</name>
```
Comment (gflores-jahia) [inline src/main/resources/META-INF/sitemap.tld:30]: see the renaming comment on VanityUrls class

**Review by gflores-jahia — CHANGES_REQUESTED** [review]

File: src/main/java/org/jahia/modules/sitemap/utils/VanityUrls.java (hunk 1 above)
Comment (AKarmanov) [reply src/main/java/org/jahia/modules/sitemap/utils/VanityUrls.java:60]: Yes, you're right there is

File: src/main/java/org/jahia/modules/sitemap/utils/VanityUrls.java (hunk 2 above)
Comment (AKarmanov) [reply src/main/java/org/jahia/modules/sitemap/utils/VanityUrls.java:59]: That makes sense.

File: src/main/resources/common/sitemap-entry.jsp [hunk 4]
```diff
@@ -2,15 +2,23 @@
 <%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
 <%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
 <%@ taglib prefix="sitemap" uri="http://www.jahia.org/sitemap" %>
+<%@ taglib prefix="fn" uri="http://java.sun.com/jsp/jstl/functions" %>
+
+<%--@elvariable id="url" type="org.jahia.services.render.URLGenerator"--%>
 
 <c:set var="renderContext" value="${requestScope['renderContext']}"/>
 <c:set var="urlNode" value="${requestScope['urlNode']}"/>
+<c:set var="canUseVanity" value="${sitemap:urlRewriteEnabled()}"/>
```
Comment (gflores-jahia) [inline src/main/resources/common/sitemap-entry.jsp:11]: I'm sorry I'm going to be a pain and ask if we can rename the variable here as well to match the function name

**Review by gflores-jahia — CHANGES_REQUESTED** [review]

**Review by gflores-jahia — APPROVED** [review]
